conjurapi/authn/auth_token_test.go

```diff
@@ -64,3 +73,74 @@ func TestTokenV5_Parse(t *testing.T) {
 		So(err.Error(), ShouldEqual, "Unable to unmarshal v5 access token field 'payload' : invalid character 'o' in literal false (expecting 'a')")
 	})
 }
+
+func TestTokenV4_Parse(t *testing.T) {
+	expired_token_bytes := []byte(`{"data":"admin","timestamp":"2018-04-06 03:10:08 UTC","signature":"QxTMoWWYXbgMo_JuX4KHQuiPwPRe8fpIlnZMhlvHalyhJHK0RbkqOyw28ImLwClBaTPjx6KU7KmqYLi9pMszHQZhQ7A2fLm1v-x0XzZGrDOt6gd0fTEZ0CJl7VVxVBZWLrJ83r8tY-sdjKysrE1fyDXyMU_vDtgJVi9y72qddkH-Pl16Pd4PJceEEybfWylIs1Z5V5qn-ocWX18D-i9pB67Usz3m-wKa43TptiDYLGU1-Y_EXyilv_uNGouqwYa0IueK5yJxO1Rcyb2aCBG0i-0Vl7qYrT0zIwDqmxLAwbqOtrtfHngFOCqsW04jJLPOruR5FwMlGw90GT1lZH_3GCm6QK8p15IWfVS9UOky8Y4l-1vfh-d15BZPGemUbu0j","key":"86ffd9d612ad06fe978b559fbeba4ca2"}`)
+	var expired_token *AuthnToken4
+
+	Convey("Token type V4 is detected", t, func() {
+		token, err := NewToken(expired_token_bytes)
+		err = token.FromJSON(expired_token_bytes)
+
+		So(err, ShouldBeNil)
+		So(reflect.TypeOf(token).String(), ShouldEqual, "*authn.AuthnToken4")
+		So(token.Raw(), ShouldNotBeNil)
+
+		expired_token, _ = token.(*AuthnToken4)
+	})
+
+	Convey("Token timestamp is non-zero", t, func() {
+		So(expired_token.Timestamp.IsZero(), ShouldEqual, false)
+	})
+
+	Convey("Expired token should be refreshed", t, func() {
+		So(expired_token.ShouldRefresh(), ShouldEqual, true)
+	})
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+
+	httpClient := http.Client{Transport: tr}
```

Comment:
Unfortunately we can't import `conjurapi` from here so I had to just make a raw HTTP request to get a new token.